src/ir_Transcold.cpp

```diff
@@ -1,93 +1,542 @@
-  if (strict && nbits != kTranscoldBits)
+                          const uint16_t nbits, const bool strict) {
+  // The protocol sends the data normal + inverted, alternating on
+  // each byte. Hence twice the number of expected data bits.
+  if (results->rawlen <= 2 * 2 * nbits + kHeader + kFooter - 1 + offset)
     return false;
+  if (strict && nbits != kTranscoldBits) return false;
+  if (nbits % 8 != 0) return false;
 
   uint64_t data = 0;
+  uint64_t inverted = 0;
 
-  // Match Header + Data + Footer (1 of 2)
-  uint16_t used = matchGeneric(results->rawbuf + offset, &data,
-                               results->rawlen - offset, nbits,
-                               // Header
-                               kTranscoldHdrMark, kTranscoldHdrSpace,
-                               // Data
-                               kTranscoldBitMark, kTranscoldOneSpace,
-                               kTranscoldBitMark, kTranscoldZeroSpace,
-                               // Footer (1 of 2)
-                               kTranscoldBitMark, kTranscoldHdrSpace,
-                               false, _tolerance, 0, true);
-  if (!used) return false;  // Didn't fully match.
-  offset += used;
-
-  // Footer (2 of 2)
-  if (!matchMark(results->rawbuf[offset++], kTranscoldBitMark))
-    return false;
-  if (offset <= results->rawlen &&
+  if (nbits > sizeof(data) * 8)
+    return false;  // We can't possibly capture a Transcold packet that big.
+
+  // Header
+  if (!matchMark(results->rawbuf[offset++], kTranscoldHdrMark)) return false;
+  if (!matchSpace(results->rawbuf[offset++], kTranscoldHdrSpace)) return false;
+
+  // Data
+  // Twice as many bits as there are normal plus inverted bits.
+  for (uint16_t i = 0; i < nbits * 2; i++, offset++) {
+    bool flip = (i / 8) % 2;
+    if (!matchMark(results->rawbuf[offset++], kTranscoldBitMark))
+      return false;
+    if (matchSpace(results->rawbuf[offset], kTranscoldOneSpace)) {
+      if (flip)
+        inverted = (inverted << 1) | 1;
+      else
+        data = (data << 1) | 1;
+    } else if (matchSpace(results->rawbuf[offset], kTranscoldZeroSpace)) {
+      if (flip)
+        inverted <<= 1;
+      else
+        data <<= 1;
+    } else {
+      return false;
+    }
+  }
+
+  // Footer
+  if (!matchMark(results->rawbuf[offset++], kTranscoldBitMark)) return false;
+  if (!matchSpace(results->rawbuf[offset++], kTranscoldHdrSpace)) return false;
+  if (!matchMark(results->rawbuf[offset++], kTranscoldBitMark)) return false;
+  if (offset < results->rawlen &&
       !matchAtLeast(results->rawbuf[offset], kDefaultMessageGap))
     return false;
 
+  // Compliance
+  if (strict && inverted != invertBits(data, nbits)) return false;
+
   // Success
   results->decode_type = decode_type_t::TRANSCOLD;
   results->bits = nbits;
   results->value = data;
-  results->command = 0;
   results->address = 0;
+  results->command = 0;
   return true;
 }
-#endif  // DECODE_TRANSCOLD
+#endif
```

Comment:
`  // DECODE_TRANSCOLD` removed?

Reply:
D'oh. I blindly cut-n-pasted the code. My bad. Fixed.